airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_streams.py

```diff
@@ -0,0 +1,159 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from json import dumps, loads
+from urllib.parse import parse_qs, urlparse
+
+import pytest
+import responses
+from airbyte_cdk.models import SyncMode
+from jsonschema import validate
+from source_amazon_ads import SourceAmazonAds
+
+
+def setup_responses(profiles_response=None, campaigns_response=None):
+    responses.add(
+        responses.POST,
+        "https://api.amazon.com/auth/o2/token",
+        json={"access_token": "alala", "expires_in": 10},
+    )
+    if profiles_response:
+        responses.add(
+            responses.GET,
+            "https://advertising-api.amazon.com/v2/profiles",
+            body=profiles_response,
+        )
+    if campaigns_response:
+        responses.add(
+            responses.GET,
+            "https://advertising-api.amazon.com/sd/campaigns",
+            body=campaigns_response,
+        )
+
+
+@responses.activate
+def test_streams_profile(test_config, profiles_response):
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    assert len(streams) == 6
```

Comment:
Why are you checking streams length in `test_streams_profile()` and not in `test_streams_campaigns_one_vendor()`, for example, or in `test_streams_campaigns_4_vendors()`?  I mean I know why, because you added `test_streams_profile()` first. And don't you see here a problem?